go-chaos/cmd/cluster.go

```diff
@@ -0,0 +1,190 @@
+)
+
+func AddClusterCommands(rootCmd *cobra.Command, flags *Flags) {
+	var clusterCommand = &cobra.Command{
+		Use:   "cluster",
+		Short: "Interact with the Cluster API",
+		Long:  "Can be used query cluster topology and to request dynamic scaling",
+	}
+	var statusCommand = &cobra.Command{
+		Use:   "status",
+		Short: "Queries the current cluster topology",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return printCurrentTopology(flags)
+		},
+	}
+	var waitCommand = &cobra.Command{
+		Use:   "wait",
+		Short: "Waits for a topology change to complete",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return waitForChange(flags)
+		},
+	}
+
+	rootCmd.AddCommand(clusterCommand)
+	clusterCommand.AddCommand(statusCommand)
+	clusterCommand.AddCommand(waitCommand)
+	waitCommand.Flags().IntVar(&flags.changeId, "changeId", 0, "The id of the change to wait for")
+}
+
+func printCurrentTopology(flags *Flags) error {
+	k8Client, err := createK8ClientWithFlags(flags)
+	if err != nil {
+		panic(err)
+	}
+
+	err = k8Client.AwaitReadiness()
+	if err != nil {
+		return err
+	}
+
+	port := 9600
+	closePortForward := k8Client.MustGatewayPortForward(port, port)
+	defer closePortForward()
+
+	topology, err := queryTopology(port)
+	if err != nil {
+		return err
+	}
+	formatted, err := json.MarshalIndent(topology, "", "  ")
+	if err != nil {
+		return err
+	}
+	fmt.Println(string(formatted))
+	return nil
+}
+
+func waitForChange(flags *Flags) error {
+	k8Client, err := createK8ClientWithFlags(flags)
+	if err != nil {
+		panic(err)
+	}
+
+	err = k8Client.AwaitReadiness()
+	if err != nil {
+		return err
+	}
+
+	port := 9600
+	closePortForward := k8Client.MustGatewayPortForward(port, port)
+	defer closePortForward()
+
+	for {
+		topology, err := queryTopology(port)
+		if err != nil {
+			return err
+		}
+		if topology.LastChange != nil && topology.LastChange.Id == int64(flags.changeId) {
+			if topology.LastChange.Status == "COMPLETED" {
+				internal.LogInfo("Change %d completed successfully", flags.changeId)
+				return nil
+			} else {
+				return fmt.Errorf("change %d failed with status %s", flags.changeId, topology.LastChange.Status)
+			}
+		} else if topology.LastChange != nil && topology.LastChange.Id > int64(flags.changeId) {
+			internal.LogInfo("Change %d is outdated but was most likely completed successfully, latest change is %d", flags.changeId, topology.LastChange.Id)
+			return nil
+		} else if topology.PendingChange != nil && topology.PendingChange.Id == int64(flags.changeId) {
+			competed := len(topology.PendingChange.Completed)
+			pending := len(topology.PendingChange.Pending)
+			total := competed + pending
+			internal.LogInfo("Change %d is %s with %d/%d operations complete", flags.changeId, topology.PendingChange.Status, competed, total)
+		} else {
+			internal.LogInfo("Change %d not yet started", flags.changeId)
+		}
+
+		time.Sleep(5 * time.Second)
+	}
+}
+
+func queryTopology(port int) (*CurrentTopology, error) {
```

Comment:
🔧 Would be great if you could add a IT for it

Reply:
I've added a minimal one now. None of the other commands have integration tests so I hope I did it in a way that is acceptable: https://github.com/zeebe-io/zeebe-chaos/pull/459/commits/ffd0a20c687e7616b605de8be0edf687e0c5a8f1

Reply:
No integration tests for other commands you say? 

![shock](https://media.giphy.com/media/l2RsBwQxFPXUvXmi0u/giphy-downsized.gif)